packages/@aws-cdk/aws-cloudfront/test/test.basic.ts

```diff
@@ -322,4 +323,168 @@ export = {
     test.done();
   },
 
+  'distribution with resolvable lambda-association'(test: Test) {
+    const stack = new cdk.Stack();
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+
+    const lambdaFunction = new lambda.Function(stack, 'Lambda', {
+      code: lambda.Code.inline('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.Nodejs810
+    });
+
+    const lambdaVersion = new lambda.Version(stack, 'LambdaVersion', {
+      lambda: lambdaFunction
+    });
+
+    new CloudFrontWebDistribution(stack, 'AnAmazingWebsiteProbably', {
+      originConfigs: [
+        {
+          s3OriginSource: {
+            s3BucketSource: sourceBucket
+          },
+          behaviors: [
+            {
+              isDefaultBehavior: true,
+              lambdaFunctionAssociations: [{
+                eventType: LambdaEdgeEventType.OriginRequest,
+                lambdaFunction: cdk.Token.asAny(lambdaVersion.versionArn)
```

Comment:
I don't understand why the `Token.asAny()` is necessary.

Can you explain to me why:

```
lambdaFunction: lambdaVersion
```

Is not good enough?

Reply:
This test should test the case that `lambdaFunction` is a token. Therefore the `lambdaVersion` which is an `IVersion` is not good enought and I kind of cast the versionARN to a token.
This is semantically and in types of typesafety correct for this test case.

Reply:
A bit deeper explanation is this:
If you have a cloudfront distribution in a region other than `us-east-1` and want a lambda@edge to use with that, then you need to do the following:

Deploy your lambda in region `us-east-1`, create a custom-resource to fetch the lambda version ARN and use the custom resource (which is in case of CDK a token) as function association with your distribution in your region.

Reply:
Aha, so you have an ARN and you need to turn it into an `IVersion` so that you can pass it in the right place?

Normally, we have a function like `Version.fromVersionArn()` for that, similar to `Bucket.fromBucket(Name,Arn)`, and others.

The input of `Version.fromVersionArn()` could be the output of your custom resource.

Reply:
Its not even sure that the value is an ARN. Its only like this in the test case for simplicity.
In a real case one creates a custom resoures and gets an output from it like this:

```ts
    const stackOutput = new cloudformation.CustomResource(this, 'StackOutput', {
      provider: cloudformation.CustomResourceProvider.lambda(
        lambdaEdgeVersionProvider
      ),
      properties: {
        StackName: props.edgeStackName,
        OutputKey: 'Output'
        // just to change custom resource on code update
        // LambdaHash: sha256('./lambda/index.js')
      }
    });

    return stackOutput.getAtt('Output');
```
The result of this is `import("@aws-cdk/cdk").CfnReference;` which is a token. I'm not sure how to create a version ARN from this reference.

Reply:
Hmm. That attribute should have been an `IResolvable`, so if it's not that's a bug.


> Its not even sure that the value is an ARN

Let me put it to you like this:

The only way that code makes sense is if the output of that custom resource is actually a function ARN (I suppose a function name + version would also work, but that's just a shorter way to write the same information). Phrased another way, if the output of the Custom Resource was a bucket name, you wouldn't expect to be able to pass that into the `lambdaFunction` argument and have it do something useful. So in a sense, the output of your custom resource is definitely something that indicates a Lambda Version, so it makes sense that you would be able to "convert" that into an `IVersion` object.

The solution is to create either/both `Version.fromVersionArn()`/`Version.fromVersionQualifier()` in order to make the rest of this code work.

Reply:
Got it, I'll update the code to be like that. Thanks

Reply:
@rix0rrr Unfortunately I could not get this to work. I've updated the test to use a CustomResource and get the `attr(...)`. This is as string representation `${Token[Default.Output.69]}` which could not get parsed as ARN. So I'm not sure how to create an ARN from a reference which is the output of the custom resource. Maybe I'm not deep enough in the CDK...

Reply:
Oh, never mind. There was something missing in the imported version. I'll can fix that now and finish this PR. :smile: